nupic/research/frameworks/dendrites/dendritic_layers.py

```diff
@@ -102,16 +102,12 @@ class AbsoluteMaxGatingDendriticLayer(BiasingDendriticLayer):
     """
 
     def apply_dendrites(self, y, dendrite_activations):
-        output_max = dendrite_activations.max(dim=2).values
-        output_min = dendrite_activations.min(dim=2).values
-        sign_mask = torch.where(
-            (output_max ** 2) > (output_min ** 2),
-            torch.ones(output_max.shape).to(y.device),
-            -1.0 * torch.ones(output_min.shape).to(y.device)
-        )
-        return y * torch.sigmoid(
-            torch.abs(dendrite_activations).max(dim=2).values * sign_mask
-        )
+        inds = dendrite_activations.abs().max(dim=2).indices
+        inds = inds.unsqueeze(dim=2)
+        dendrite_activations = torch.gather(dendrite_activations, dim=2, index=inds)
+        dendrite_activations = dendrite_activations.squeeze()
+        dendrite_activations = torch.sigmoid(dendrite_activations)
+        return y * dendrite_activations
```

Comment:
This looks good, definitely. Did you time this method to compare it to the previous one?

Reply:
I didn't, but I'll do that (probably after I merge) and will get back to you about it.

Reply:
Thanks, please do. It's always good to do a sanity check on these sorts of things. Indices especially have many ways to use them, and they each could have their own tradeoffs with respect to time and memory resources.